examples/request_version/request_version.cc

```diff
@@ -0,0 +1,116 @@
+/**************************************************************************/ /**
+* @brief Message encode example.
+* @file
+******************************************************************************/
+
+#include <cmath>
+#include <cstdio>
+#include <cstring>
+
+#include <point_one/fusion_engine/messages/core.h>
+#include <point_one/fusion_engine/messages/crc.h>
+#include <point_one/fusion_engine/parsers/fusion_engine_framer.h>
+
+#include "../common/print_message.h"
+
+using namespace point_one::fusion_engine::examples;
+using namespace point_one::fusion_engine::messages;
+using namespace point_one::fusion_engine::parsers;
+
+bool message_found = false;
+
+int main(int argc, const char* argv[]) {
+  if (argc != 1) {
+    printf("Usage: %s\n", argv[0]);
+    printf(R"EOF(
+Simulate sending a version request, and parsing the response.
+)EOF");
+    return 0;
+  }
+
+  // Enforce a 4-byte aligned address.
+  alignas(4) uint8_t storage[4096];
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Write a VersionInfoMessage request.
+  //////////////////////////////////////////////////////////////////////////////
+
+  uint8_t* buffer = storage;
+  auto header = reinterpret_cast<MessageHeader*>(buffer);
+  buffer += sizeof(MessageHeader);
+  *header = MessageHeader();
+
+  header->sequence_number = 0;
+  header->message_type = MessageType::MESSAGE_REQUEST;
+  header->payload_size_bytes = sizeof(MessageRequest);
+
+  auto req_message = reinterpret_cast<MessageRequest*>(buffer);
+  *req_message = MessageRequest();
+
+  req_message->message_type = VersionInfoMessage::MESSAGE_TYPE;
+
+  header->crc = CalculateCRC(storage);
+
+  printf("Sending VersionInfoMessage request:\n  ");
+  // This data would be sent over serial to the device.
+  PrintHex(storage, sizeof(MessageHeader) + sizeof(MessageRequest));
+  printf("\n");
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Generate an example response
```

Comment:
I know you make the comment later, but its probably worth explaining up here that you're generating this example response that would normally be generated by the hardware for clarity.